site/docs/pt/guide/middleware.md

```diff
@@ -0,0 +1,258 @@
+
+::: code-group
+
+```ts [TypeScript]
+/** Mede o tempo de resposta do bot e o registra no `console` */
+async function responseTime(
+  ctx: Context,
+  next: NextFunction, // é um alias para: () => Promise<void>
+): Promise<void> {
+  // TODO: implementar
+}
+```
+
+```js [JavaScript]
+/** Measures the response time of the bot, and logs it to `console` */
+/** Mede o tempo de resposta do bot e o registra no `console` */
+async function responseTime(ctx, next) {
+  // TODO: implementar
+}
+```
+
+:::
+
+Podemos instalá-lo em nossa instância `bot` com `bot.use()`:
+
+```ts
+bot.use(responseTime);
+```
+
+Vamos começar a implementá-lo.
+Aqui está o que queremos fazer:
+
+1. Assim que um update chegar, armazenamos `Date.now()` em uma variável.
+2. Invocamos o middleware downstream, portanto, deixamos todo o tratamento de mensagens acontecer.
+   Isso inclui a busca por comandos, respostas e tudo mais que seu bot faz.
+3. Pegamos `Date.now()` novamente, comparamos com o valor antigo e chamamos `console.log` com a diferença de tempo.
+
+É importante instalar nosso middleware `responseTime` _primeiro_ no bot (no topo da pilha de middlewares) para garantir que todas as operações sejam incluídas na medição.
+
+::: code-group
+
+```ts [TypeScript]
+/** Mede o tempo de resposta do bot e o registra no `console` */
+async function responseTime(
+  ctx: Context,
+  next: NextFunction, // é um alias alias para: () => Promise<void>
+): Promise<void> {
+  // obtém a hora antes
+  const before = Date.now(); // millissegundos
+  // invoca o middleware downstream
+  await next(); // certifique-se de dar `await`!
+  // obtém a hora depois
+  const after = Date.now(); // millissegundos
+  // registra a diferença
+  console.log(`Tempo de resposta: ${after - before} ms`);
+}
+
+bot.use(responseTime);
+```
+
+```js [JavaScript]
+/** Mede o tempo de resposta do bot e o registra no `console` */
+async function responseTime(ctx, next) {
+  // obtém a hora antes
+  const before = Date.now(); // millisegundos
+  // invoca o middleware downstream
+  await next(); // certifique-se de dar `await`!
+  // obtém a hora depois
+  const after = Date.now(); // millissegundos
+  // registra a diferença
+  console.log(`Response time: ${after - before} ms`);
+}
+
+bot.use(responseTime);
+```
+
+:::
+
+Completo e funciona! :heavy_check_mark:
+
+Sinta-se à vontade para usar este middleware no seu objeto bot, registrar mais listeners e brincar com o exemplo.
+Fazê-lo ajudará você a entender completamente o que é middleware.
+
+::: danger PERIGO: Sempre certifique-se de dar await no next!
+Se você chamar `next()` sem a palavra-chave `await`, várias coisas vão quebrar:
+
+- :x: Sua pilha de middlewares será executada na ordem errada.
+- :x: Você pode perder dados.
+- :x: Algumas mensagens podem não ser enviadas.
+- :x: Seu bot pode travar aleatoriamente de maneiras difíceis de reproduzir.
```

Comment:
```suggestion
- :x: Seu bot pode quebrar aleatoriamente de maneiras difíceis de reproduzir.
```
It's not freezing, it's crashing. Maybe "parar" can be used too. But I think "quebrar" express better the idea